app/server/src/main/scala/org/bitcoins/server/NodeRoutes.scala

```diff
@@ -8,12 +8,22 @@ import org.bitcoins.node.Node
 
 case class NodeRoutes(node: Node)(implicit system: ActorSystem)
     extends ServerRoute {
-  implicit val materializer = ActorMaterializer()
+  import system.dispatcher
+  implicit val materializer: ActorMaterializer = ActorMaterializer()
 
   def handleCommand: PartialFunction[ServerCommand, StandardRoute] = {
     case ServerCommand("getpeers", _) =>
       complete {
         Server.httpSuccess("TODO implement getpeers")
       }
+
+    case ServerCommand("stop", _) =>
+      complete {
+        node.stop().map { _ =>
```

Comment:
So a couple things here: 

1. If `node.stop()` fails, nothing more is shutdown. I don't think we want thie behavior.
2. This `stop` command will return before the server is actually shutdown. Is this what we want? I don't think we can serve a response if we shutdown the server first, so it's a weird catch 22. I think we want a `Await.result()` somewhere, but i'm not sure where.
3. This probably should be implemented in `ServerRoute`, so this can be used across all projects. We don't want this just available in `NodeRoutes`.

Reply:
1. Agreed
2. If we add an `Await.result()` it needs to be outside the `complete` so the cli doesn't get a "Request failed: Future timed out after [X seconds]". However, `complete` is not a future, it runs asynchronously AFAIK so I'm not sure if this possible
3. I thought about doing this originally but not sure if it really makes sense in practicality. We have 4 different `XRoutes`: Node, Core, Chain, and Wallet. Node is the only one here that is actually an active process, the rest are just single api calls that update a database, there is no running process to actually stop.